### PR TITLE
Remove strict comparison of tensors against golden values in evo2 test

### PR DESCRIPTION
**Pull request by balvisio**
### Description

The pytorch 25.04 has CUDA 12.9 installed:
```
Build cuda_12.9.r12.9/compiler.35813241_0
```

while pytorch 25.01 has 12.8
```
Cuda compilation tools, release 12.8, V12.8.61
Build cuda_12.8.r12.8/compiler.35404655_0
```

This seems to be causing numerical differences so that strict checking of tensor values fails.

### Type of changes

- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  New feature (non-breaking change which adds functionality)
- [ ]  Refactor
- [ ]  Documentation update
- [x]  Other (please describe):
Remove a strict tensor comparison that doesn't hold across CUDA version.

### CI Pipeline Configuration
Configure CI behavior by applying the relevant labels:

- [SKIP_CI](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#skip_ci) - Skip all continuous integration tests
- [INCLUDE_NOTEBOOKS_TESTS](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#include_notebooks_tests) - Execute notebook validation tests in pytest
- [INCLUDE_SLOW_TESTS](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#include_slow_tests) - Execute tests labelled as slow in pytest for extensive testing

> [!NOTE]
> By default, the notebooks validation tests are skipped unless explicitly enabled.

#### Authorizing CI Runs

We use [copy-pr-bot](https://docs.gha-runners.nvidia.com/apps/copy-pr-bot/#automation) to manage authorization of CI
runs on NVIDIA's compute resources.

* If a pull request is opened by a trusted user and contains only trusted changes, the pull request's code will
  automatically be copied to a pull-request/ prefixed branch in the source repository (e.g. pull-request/123)
* If a pull request is opened by an untrusted user or contains untrusted changes, an NVIDIA org member must leave an
  `/ok to test` comment on the pull request to trigger CI. This will need to be done for each new commit.

### Usage

```python
TODO: Add code snippet
```

### Pre-submit Checklist

 - [x] I have tested these changes locally
 - [x] I have updated the documentation accordingly
 - [x] I have added/updated tests as needed
 - [ ] All existing tests pass successfully

### CODE REVIEW

**Review by jstjohn — COMMENTED** [review]
Please see https://github.com/NVIDIA/bionemo-framework/pull/905 which I think should also land. We need to check that changes work with a downstream task such as AUC on BRCA1. cc @dorotat-nv @trvachov.

**Comment by balvisio** [comment]
This PR is adding a downstream task assertion on performance https://github.com/NVIDIA/bionemo-framework/pull/905

**Comment by balvisio** [comment]
@trvachov and @dorotat-nv ping for review. Thank you!

File: sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py [hunk 1]
```diff
@@ -175,19 +174,6 @@ def test_golden_values_top_k_logits_and_cosine_similarity_7b(seq_len: int = 8_19
         attention_mask = None
         outputs = model(input_ids=input_ids, position_ids=position_ids, attention_mask=attention_mask)
         gold_standard_no_fp8_tensor = torch.load(gold_standard_no_fp8).to(device=outputs.device, dtype=outputs.dtype)
-        is_fp8_supported, compute_capability, device_info = check_fp8_support(device.index)
```
Comment (dorotat-nv) [inline sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py:178]: is there any case where the strict test is satisfied, ie A100 on x86? maybe instead of removing this check entirely, we should scope the if statement or add some tolerance?

Comment (dorotat-nv) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py:178]: cc @jstjohn

Comment (jstjohn) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py:178]: I thought the problem was that the torch upgrade made it so that exact equality stopped working across the board. Is that the case @balvisio or should we instead change this to exact if the device is older than a b100 but still fp8, eg L40 or H100 would be ok?

Comment (balvisio) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py:178]: The exact equality stops working on both x86 and arm64. I have only tested this on H200 since that is where the nightly runs. Haven't tried other GPUs.

Comment (balvisio) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py:178]: @jstjohn , @dorotat-nv : do we want to perform tests to confirm it happens on other GPU models too or you think this is ready?

Comment (jstjohn) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py:178]: I'm ok, between this and the AUC test we added into the notebook we're probably ok.

**Review by jstjohn — APPROVED** [review]
